python/revng/internal/cli/_commands/generate_migrations/__init__.py

```diff
@@ -0,0 +1,229 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import sys
+import subprocess
+from pathlib import Path
+from typing import Dict, List, Literal, Tuple, Any
+
+import jinja2
+import jsonschema
+import yaml
+
+from revng.internal.cli.commands_registry import Command, CommandsRegistry
+from revng.internal.cli.support import log_error
+
+Change = (
+    Tuple[Literal["definition"], Literal["add", "remove"], str]
+    | Tuple[Literal["definition"], Literal["edit"], str, str]
+    | Tuple[Literal["sub_definition"], Literal["add", "remove"], str, str]
+    | Tuple[Literal["sub_definition"], Literal["edit"], str, str, str]
+)
+
+class GenerateMigrationCommand(Command):
+    def __init__(self):
+        super().__init__(
+            ("model", "generate-migrations"),
+            "Generate migrations for a schema based on its git history",
+        )
+
+    def register_arguments(self, parser):
+        parser.add_argument("schema", help="The path to the schema.")
+        parser.add_argument("upstream_branch", help="The name of the upstream branch where the previous version of the schema will be taken from.")
```

Comment:
1. This line seems very long.
2. Drop any reference to "branch", talk about "commit" instead.